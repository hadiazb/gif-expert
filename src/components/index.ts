export * from './GifItem';
export * from './gifSearch.interface';
export * from './GifGrid';
export * from './AddCategory';
