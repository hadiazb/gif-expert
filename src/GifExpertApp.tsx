import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState<string[]>([]);

	const onAddCategory = (newCategory: string) => {
		if (categories.includes(newCategory)) return;
		setCategories((categories) => [newCategory, ...categories]);
	};

	return (
		<>
			<h1>GifExpertApp</h1>
			<AddCategory onNewCategory={(event) => onAddCategory(event)} />
			<ol>
				{categories.length > 0 &&
					categories.map((category) => (
						<GifGrid key={category} category={category} />
					))}
			</ol>
		</>
	);
};
