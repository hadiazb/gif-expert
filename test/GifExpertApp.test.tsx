import { render, screen } from '@testing-library/react';

import { GifExpertApp } from '../src/GifExpertApp';

describe('GifExpertApp', () => {
	test('should first', () => {
		const { container } = render(<GifExpertApp />);

		expect(container).toMatchSnapshot();
	});
});
