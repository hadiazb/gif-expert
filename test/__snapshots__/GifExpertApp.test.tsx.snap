// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should first 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <ol />
</div>
`;
